app/src/components/security/RouteGuards.tsx

```diff
@@ -119,26 +119,14 @@ export const UnAuthenticatedRouteGuard = (props: RouteProps) => {
  * @param {*} { children }
  * @return {*}
  */
-const CheckForAuthLoginParam = (props: PropsWithChildren<Record<never, unknown>>) => {
+const CheckForKeycloakAuthenticated = (props: PropsWithChildren<Record<never, unknown>>) => {
   const { keycloakWrapper } = useContext(AuthStateContext);
 
   const location = useLocation();
 
   if (!keycloakWrapper?.keycloak?.authenticated) {
-    const urlParams = qs.parse(location.search.replace('?', ''));
-    const authLoginUrlParam = urlParams.authLogin;
-
-    // Check for urlParam to force login
-    if (authLoginUrlParam) {
-      // Remove authLogin url param from url to stop possible loop redirect
-      const redirectUrlParams = qs.stringify(urlParams, { filter: (prefix) => prefix !== 'authLogin' });
-      const redirectUri = `${window.location.origin}${location.pathname}?${redirectUrlParams}`;
-
-      // Trigger login
-      keycloakWrapper?.keycloak?.login({ redirectUri: redirectUri });
-    }
-
-    return <Redirect to="/" />;
+    // Trigger login, then redirect to the desired route
+    return <Redirect to={`/login?next=${encodeURIComponent(location.pathname)}`} />;;
```

Comment:
nit: the keycloak and siteminder redirects use `redirect_uri` and `returl` respectively. Maybe we can change `next` to something more similar: `redirect` or `redirect_uri` seem like good options, just so its more clear what this field is for.

Reply:
Changed to `redirect`.